### DISCUSSION TITLE
Create a 'developer' or 'debug' build variant

### DISCUSSION THREAD

**Discussion by kvithayathil** [Docile-Alligator/Infinity-For-Reddit#1122] [Ideas]
Hi,

I've been exploring the code and I feel like it's a great project to contribute to.
When I was building the app I noticed there's no distinction between a local build and a production build. It would be nice to have a 'developer'/'debug' variant to build against when doing local developing. 

We can utilise features in gradle to allow for side-by-side installation of the debug and store version. For example, we can append something to the package name to prevent namespace clashes upon installation.

```groovy
    flavorDimensions 'BuildVariant'
    productFlavors {
        develop {
            dimension 'BuildVariant'
            applicationIdSuffix '.dev'
        }
        production {
            dimension 'BuildVariant'
        }
```

We can also use this to provide separate app names for build types e.g. 'Infinity-dev' and even separate assets such as the logo. I provided a rough example below:

![infinity_dev_logo_test](https://user-images.githubusercontent.com/1056073/191292824-8c728513-2546-4900-9e90-49e09ad82cfc.png)

Let me know your thoughts, I appreciate your efforts on this project!

**Comment by Wladefant** [comment]
Yes, we are already thinking about it and there is even a pull request for that: #636

**Reply by kvithayathil** [reply]
Awesome, I'll take a look
